apps/www/_blog/2024-09-23-local-first-expo-legend-state.mdx

```diff
@@ -0,0 +1,346 @@
+
+1. Save the pending changes to local persistence.
+1. Save the changes to local persistence.
+1. Save the changes to remote persistence.
+1. On remote save, set any needed changes (like updatedAt) back into the observable and local persistence.
+1. Clear the pending changes in local persistence.
+
+## Setting up the Project
+
+To set up a new React Native project you can use the `create-expo-app` utility. You can create a blank app or choose from different [examples](https://github.com/expo/examples).
+
+For this tutorial, go ahead and create a new blank Expo app:
+
+```bash
+npx create-expo-app@latest --template blank
+```
+
+## Installing Dependencies
+
+The main dependencies you need are [Legend State](https://www.npmjs.com/package/@legendapp/state) and [supabase-js](https://www.npmjs.com/package/@supabase/supabase-js). Additionally, to make things work for React Native, you will need [react-native-mmkv](https://github.com/mrousavy/react-native-mmkv) and [react-native-get-random-values](https://www.npmjs.com/package/react-native-get-random-values) (to generate uuids).
+
+Install the required dependencies via `expo install`:
+
+```bash
+npx expo install @legendapp/state @supabase/supabase-js react-native-get-random-values react-native-mmkv
+```
+
+## Configuring Supabase
+
+If you don't have a Supabase project already, head over to [database.new](https://database.new) and create a new project.
+
+Next, create a `.env.local` file in the root of your project and add the following env vars. You can find these in your [Supabase dashboard](https://supabase.com/dashboard/project/_/settings/api).
+
+```bash
+EXPO_PUBLIC_SUPABASE_URL=
+EXPO_PUBLIC_SUPABASE_ANON_KEY=
+```
+
+Next, set up a utils file to hold all the logic for interacting with Supabase, we'll call it `utils/SupaLegend.ts`.
+
+```ts utils/SupaLegend.ts
+import { createClient } from '@supabase/supabase-js'
+
+const supabase = createClient(
+  process.env.EXPO_PUBLIC_SUPABASE_URL,
+  process.env.EXPO_PUBLIC_SUPABASE_ANON_KEY
+)
+```
+
+## Configuring Legend-State
+
+Legend-State is very versatile and allows you to choose different persistence and storage strategies. For this example, we'll use `MMKV` for local persistence across React Native platforms and `supabase` for remote persistence.
+
+Extend your `utils/SupaLegend.ts` file with the following configuration:
+
+```ts utils/SupaLegend.ts
+import { createClient } from '@supabase/supabase-js'
+import { observable } from '@legendapp/state'
+import { syncedSupabase } from '@legendapp/state/sync-plugins/supabase'
+import { configureSynced } from '@legendapp/state/sync'
+import { ObservablePersistMMKV } from '@legendapp/state/persist-plugins/mmkv'
+
+const supabase = createClient(
+  process.env.EXPO_PUBLIC_SUPABASE_URL,
+  process.env.EXPO_PUBLIC_SUPABASE_ANON_KEY
+)
+
+// Create a configured sync function
+const customSynced = configureSynced(syncedSupabase, {
```

Comment:
You may want to add some explanation that `syncedSupabase` is the plugin for supabase and this configures it with defaults?